packages/components/src/components/money/money.jsx

```diff
@@ -17,13 +18,22 @@ const Money = ({
 
     // if it's formatted already then don't make any changes unless we should remove extra -/+ signs
     const value = (has_sign || should_format) ? Math.abs(amount) : amount;
-    const final_amount = should_format ? CurrencyUtils.formatMoney(currency, value, true) : value;
-
+    const decimalPart = CurrencyUtils.getDecimalPart(amount);
+    const needsCryptoToggle = CurrencyUtils.isCryptocurrency(currency) &&
+        (decimalPart && decimalPart.length > CurrencyUtils.getDecimalPlaces(currency));
+    let final_amount = null;
```

Comment:
Please initialise `final_amount` to `value` so that on line 27 you can use `else if` for `should_format` and only reassign if `should_format` is true.